src/Common/tests/System/Collections/ICollection.NonGeneric.Tests.cs

```diff
@@ -214,11 +233,25 @@ public void ICollection_NonGeneric_CopyTo_TwoDimensionArray_ThrowsArgumentExcept
         [MemberData(nameof(ValidCollectionSizes))]
         public void ICollection_NonGeneric_CopyTo_NonZeroLowerBound(int count)
         {
+            if (!ICollection_NonGeneric_CopyTo_TestNonZeroLowerBound)
+            {
+                return;
+            }
+
             ICollection collection = NonGenericICollectionFactory(count);
-            Array arr = Array.CreateInstance(typeof(object), new int[1] { 2 }, new int[1] { 2 });
+            Array arr = Array.CreateInstance(typeof(object), new int[1] { count }, new int[1] { 2 });
             Assert.Equal(1, arr.Rank);
             Assert.Equal(2, arr.GetLowerBound(0));
-            Assert.ThrowsAny<ArgumentException>(() => collection.CopyTo(arr, 0));
+            if (ICollection_NonGeneric_CopyTo_NonZeroLowerBound_ThrowType == typeof(IndexOutOfRangeException) && count == 0)
```

Comment:
Same comment here. If the conditions by which the test are different are *really* specific (in this case because of bad quality control in the original implementation :)), it would be better to override the test in that classes tests. Mostly this is because this behavior isn't indicative of how a **correct** implementation should behave.